Core/include/Acts/TrackFitting/detail/KalmanUpdateHelpers.hpp

```diff
@@ -177,10 +177,14 @@ auto kalmanHandleNoMeasurement(
   if (surface.surfaceMaterial() != nullptr) {
     typeFlags.set(TrackStateFlag::MaterialFlag);
   }
-  if (surface.associatedDetectorElement() != nullptr) {
+  if (surface.associatedDetectorElement() != nullptr &&
```

Comment:
potentially this could also be turned into a variable in this function since we use it twice and the expression is a bit cryptic